File: android/app/src/main/java/com/adqmobile/socialnetwork/data/remote/PostService.kt
package com.adqmobile.socialnetwork.data.remote

import com.adqmobile.socialnetwork.domain.post.Post
import com.adqmobile.socialnetwork.domain.post.Reaction
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface PostService {
    @POST("post")
    suspend fun postPost(@Body post: Post): Post

    @POST("post/{id}/reaction")
    suspend fun postReaction(@Path("id") id: Long, @Body reaction: Reaction): Post

    @GET("post")
    suspend fun getPosts(): Array<Post>

    @GET("post/{id}/reaction")
    suspend fun getReactionsForPost(@Path("id") id: Long): Array<Reaction>
}

File: ios/socialnetwork/data/remote/post/PostService.swift
//
//  PostService.swift
//  socialnetwork
//
//  Created by Alan Donizete Quintiliano on 13/12/19.
//  Copyright © 2019 Alan Donizete Quintiliano. All rights reserved.
//

import Combine

class PostService {

    func postPost(post: Post) -> AnyPublisher<Post, Error> {
        let api = PostPostApi(body: post)
        return HttpClient().request(api: api)
    }

    func getPosts() -> AnyPublisher<[Post], Error> {
       let api = GetPostsApi()
       return HttpClient().request(api: api)
    }

    func reactToPost(reaction: Reaction) -> AnyPublisher<Post, Error> {
        let api = ReactToPostApi(reaction: reaction)
        return HttpClient().request(api: api)
    }
}
